### PR TITLE
Collijk/feature/allow sex subsetting of population

### PR DESCRIPTION
**Pull request by collijk**
### Description

*Category*: feature
*JIRA issue*: [MIC-3003](https://jira.ihme.washington.edu/browse/MIC-3003)

- Adds a new `'include_sex'` key to the `'population'` configuration name 
  space taking values "Male", "Female", and "Both" (consistent with GBD
  conventions).  This key allows a user to run a simulation on a sex 
  subset while still keeping the standard data in the artifact.
- Clean up docstring conventions in the population subpackage. I wrote 
  most of this stuff before the `typing` package existed, so it was still doing
  typing in the docstrings.  Also clean up some docstring formatting.
- Removed old LBWSG distribution implementation.
 

### Testing
Updated unit tests to run over all parameterizations of the population 
component.

### CODE REVIEW

File: src/vivarium_public_health/population/base_population.py [hunk 1]
```diff
@@ -27,23 +33,36 @@ class BasePopulation:
...
+                "Configuration key 'population.include_sex' must be one "
+                "of ['Male', 'Female', 'Both']. "
+                f"Provided value: {self.config.include_sex}."
+            )
+
+        source_population_structure = load_population_structure(builder)
+        self.population_data = assign_demographic_proportions(
+            source_population_structure,
+            include_sex=self.config.include_sex,
+        )
```
Comment (collijk) [inline src/vivarium_public_health/population/base_population.py:65]: Everything after this line in this file is formatting changes.

File: src/vivarium_public_health/population/base_population.py [hunk 2]
```diff
@@ -359,29 +383,3 @@ def _assign_demography_with_age_bounds(
     )
     register_simulants(simulants[["entrance_time", "age"]])
     return simulants
-
-
-def _build_population_data_table(data):
```
Comment (collijk) [inline src/vivarium_public_health/population/base_population.py:364]: Don't know why this wrapper ever existed.

File: src/vivarium_public_health/population/data_transformations.py [hunk 3]
```diff
@@ -8,67 +8,94 @@
...
 import numpy as np
 import pandas as pd
+from vivarium.framework.randomness import RandomnessStream
 
+_SORT_ORDER = ["location", "year_start", "sex", "age_start"]
 
-def assign_demographic_proportions(population_data):
-    """Calculates conditional probabilities on the provided population data for use in sampling.
+
+def assign_demographic_proportions(
```
Comment (collijk) [inline src/vivarium_public_health/population/data_transformations.py:20]: Combined docstring with `_build_population_data_table`

File: src/vivarium_public_health/population/data_transformations.py [hunk 4]
```diff
@@ -8,67 +8,94 @@
...
+            'population' : Total population estimate,
+            'P(sex, location | age, year)' : Conditional probability of sex and
+            location given age and year,
+            'P(sex, location, age | year)' : Conditional probability of sex, location,
+            and age given year,
+            'P(age | year, sex, location)' : Conditional probability of age given
+            year, sex, and location.
     """
+    population_data = population_data.copy()
+    if include_sex != "Both":
```
Comment (collijk) [inline src/vivarium_public_health/population/data_transformations.py:52]: This is the core bit that updates the probability distributions we use for sampling.

File: src/vivarium_public_health/population/data_transformations.py [hunk 5]
```diff
@@ -124,21 +152,13 @@ def rescale_binned_proportions(pop_data, age_start, age_end):
         pop_data.loc[max_bin.index, "age_end"] = age_end
         pop_data.loc[padding_bin.index, "age_start"] = age_end
 
-    return pop_data
+    return pop_data[col_order].sort_values(_SORT_ORDER).reset_index(drop=True)
```
Comment (collijk) [inline src/vivarium_public_health/population/data_transformations.py:155]: Added some sort order guarantees to fix some busted tests that were relying on unenforced implementation details.

Comment (collijk) [reply src/vivarium_public_health/population/data_transformations.py:155]: Everything past here in this file is formatting

File: src/vivarium_public_health/risks/distributions.py [hunk 6]
```diff
@@ -252,133 +252,6 @@ def __repr__(self):
         return f"DichotomousDistribution(risk={self.risk})"
 
 
-class LBWSGDistribution(PolytomousDistribution):
```
Comment (collijk) [inline src/vivarium_public_health/risks/distributions.py:255]: This older implementation was left behind.  Actual one is now in the implementations subpackage.

**Review by albrja — APPROVED** [review]

**Review by mattkappel — APPROVED** [review]

**Pull request merged**